lib/ClassUtils.lua

```diff
@@ -2,22 +2,68 @@ local tea = require(script.Parent.Parent.tea)
 local TableUtils = require(script.Parent.TableUtils)
 local ClassUtils = {}
 
-function ClassUtils.makeClass(name, constructor)
-	assert(tea.string(name), "Class name must be a string")
-	assert(tea.optional(tea.callback)(constructor), "Class constructor must be a function or nil")
+local function generateMetatable(Class)
+	return {
+		__index = Class,
+		__tostring = Class.toString,
+		__eq = Class.equals,
+		__add = Class.__add,
+		__sub = Class.__sub,
+		__mul = Class.__mul,
+		__div = Class.__div,
+		__mod = Class.__mod,
+		__pow = Class.__pow,
+		__unm = Class.__unm,
+		__concat = Class.__concat,
+		__len = Class.__len,
+		__lt = Class.__lt,
+		__le = Class.__le
+	}
+end
+
+function ClassUtils.toString(instance)
+	local string = instance.Class.name .. "("
+	local first = true
+	local keys = TableUtils.keys(instance)
+	table.sort(keys)
+	for _, key in ipairs(keys) do
+		local value = instance[key]
+		if not first then
+			string = string .. ", "
+		end
+		string = string .. key .. " = " .. tostring(value)
+		first = false
+	end
+	return string .. ")"
+end
+
+function ClassUtils.makeClass(name, constructor, include)
```

Comment:
What does `include` mean? Name and constructor are pretty obvious, but I think include could do with a comment explaining what is it (and maybe a better name - `useDefaults`?).